activerecord/test/cases/tasks/postgresql_rake_test.rb

```diff
@@ -187,14 +254,16 @@ def setup
           "adapter"  => "postgresql",
           "database" => "my-app-db"
         }
-
-        ActiveRecord::Base.stubs(:connection).returns(@connection)
-        ActiveRecord::Base.stubs(:establish_connection)
       end
 
       def test_db_retrieves_charset
-        @connection.expects(:encoding)
-        ActiveRecord::Tasks::DatabaseTasks.charset @configuration
+        ActiveRecord::Base.stub(:connection, @connection) do
+          ActiveRecord::Base.stub(:clear_active_connections!, true) do
```

Comment:
why do we need to stub `clear_active_connections!`?

Reply:
We don't! Removing.